src/ray/core_worker/reference_count.cc

```diff
@@ -462,6 +462,13 @@ void ReferenceCounter::DeleteReferenceInternal(ReferenceTable::iterator it,
       ReleaseLineageReferencesInternal(ids_to_release);
     }
 
+    if (local_raylet_client_) {
+      if (it->second.owned_by_us) {
+        local_raylet_client_->DeleteSpilledObject(
+            id, [](const Status &status, const rpc::DeleteSpilledObjectReply &reply) {});
```

Comment:
Isn't adding a new RPC for deleting an object very expensive? This could easily become the new task bottleneck.

Seems to me we should at least guard this with `should_delete_value`, or perhaps it can be piggybacked on ReleasePlasmaObject().

Reply:
Probably should re-run `ray microbenchmark` after this change.